reactive/containerd.py

```diff
@@ -169,6 +183,15 @@ def config_changed():
     # Create "dumb" context based on Config
     # to avoid triggering config.changed.
     context = dict(config())
+
+    # If juju environment variables are defined, take precedent
+    # over config.yaml.
+    # See: https://github.com/dshcherb/charm-helpers/blob/eba3742de6a7023f22778ba58fbbb0ac212d2ea6/charmhelpers/core/hookenv.py#L1455
+    environment_config = env_proxy_settings()
+    if environment_config is not None:
+        config().update(environment_config)
```

Comment:
Writing to `config()` will cause problems, see https://github.com/charmed-kubernetes/charm-docker/pull/7#discussion_r303070009